src/EFCore.PG/Query/ExpressionVisitors/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -76,33 +120,59 @@ protected override Expression VisitSubQuery(SubQueryExpression expression)
                 {
                     var containsItem = Visit(contains.Item);
                     if (containsItem != null)
-                        return new ArrayAnyExpression(containsItem, Visit(fromExpression));
+                        return new CustomArrayExpression(containsItem, Visit(fromExpression), "=", true);
                 }
             }
 
             return null;
         }
 
-        protected override Expression VisitBinary(BinaryExpression expression)
+        /// <summary>
+        /// Visits a <see cref="SubQueryExpression"/> and attempts to translate a 'LIKE ANY' or 'ILIKE ANY' expression.
+        /// </summary>
+        /// <param name="expression">The expression to visit.</param>
+        /// <returns>
+        /// A 'LIKE ANY' or 'ILIKE ANY' expression or null.
+        /// </returns>
+        [CanBeNull]
+        protected virtual Expression VisitLikeAny([NotNull] SubQueryExpression expression)
         {
-            if (expression.NodeType == ExpressionType.ArrayIndex)
+            var queryModel = expression.QueryModel;
+            var results = queryModel.ResultOperators;
+
+            if (results.Count != 1 || !(results[0] is AnyResultOperator))
+                return null;
+
+            var bodyClauses = queryModel.BodyClauses;
+
+            if (bodyClauses.Count != 1 ||
+                !(bodyClauses[0] is WhereClause whereClause) ||
+                !(whereClause.Predicate is MethodCallExpression methodCallExpression))
+                return null;
+
+            if (!(Visit(methodCallExpression.Object ?? methodCallExpression.Arguments[1]) is Expression instance))
```

Comment:
Doesn't this `Visit()` call actually generate SQL? Shouldn't it happen only after you've made sure you can translate the subquery (i.e. checking the specific methods below)?